### PR TITLE
Add decoding option to specific how to decode tag 0 or 1 into any

### PR DESCRIPTION
**Pull request by ssuriyan7**
Adds a decoding option to specify how to decode CBOR tag 0 or 1 data item into an empty interface. 
Based on the specified mode, Unmarshal can return a time.Time value or a time string in a specific format.

#### PR Was Proposed and Welcomed in Currently Open Issue

- [x] This PR was proposed and welcomed by maintainer(s) in issue #496
- [x] Closes or Updates Issue #496 

#### Checklist (for code PR only, ignore for docs PR)

- [x] Include unit tests that cover the new code
- [x] Pass all unit tests 
- [x] Pass all lint checks in CI (goimports, gosec, staticcheck, etc.)
- [x] Sign each commit with your real name and email.  
      Last line of each commit message should be in this format:  
      Signed-off-by: Firstname Lastname <firstname.lastname@example.com>
- [x] Certify the Developer's Certificate of Origin 1.1
      (see next section).

#### Certify the Developer's Certificate of Origin 1.1

- [x] By marking this item as completed, I certify 
      the Developer Certificate of Origin 1.1.

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
660 York Street, Suite 102,
San Francisco, CA 94110 USA

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

### CODE REVIEW

**Comment by fxamacker** [comment]
@ssuriyan7 Thanks for opening this PR!  I'll try to take a look this week/weekend.  :pray:

**Review by fxamacker — COMMENTED** [review]
@ssuriyan7 Thanks for opening this PR!  :+1: 

I left some suggestions to make this deterministic by preserving time zone.  This will also simplify tests by eliminating the need to have tests deal with nondeterministic output.

File: decode.go [hunk 1]
```diff
@@ -1503,6 +1538,11 @@ func (d *decoder) parse(skipSelfDescribedTag bool) (interface{}, error) { //noli
 		case 0, 1:
 			d.off = tagOff
 			tm, _, err := d.parseToTime()
+			if d.dm.timeTagToAny == TimeTagToRFC3339 {
+				return tm.Format(time.RFC3339), err
+			} else if d.dm.timeTagToAny == TimeTagToRFC3339Nano {
+				return tm.Format(time.RFC3339Nano), err
+			}
```
[the remark this answers is not in the document]
Comment (ssuriyan7) [reply decode.go]: Implemented this suggestion.

File: decode_test.go [hunk 2]
```diff
@@ -8658,3 +8659,104 @@ func TestUnmarshalWithUnrecognizedTagToAnyModeForSharedTag(t *testing.T) {
...
+			} else {
+				var format string
+				if tc.opts.TimeTagToAny == TimeTagToRFC3339 {
+					format = time.RFC3339
+				} else if tc.opts.TimeTagToAny == TimeTagToRFC3339Nano {
+					format = time.RFC3339Nano
+				}
+				timeString, _ := got.(string)
+				tm, _ := time.Parse(format, timeString) // convert the string to a time value. This is needed to avoid flaky tests because time.Format can return local time strings.
+				compareNonFloats(t, tc.in, tm, tc.want)
```
[the remark this answers is not in the document]
Comment (ssuriyan7) [reply decode_test.go]: Simplified the tests.

**Comment by ssuriyan7** [comment]
@fxamacker , Thanks for your suggestion. I've incorporated them. Please take a look.

**Review by fxamacker — APPROVED** [review]
LGTM!  Thanks for contributing this PR!

**Pull request merged**